### PR TITLE
Make skip navigation foreign key nullable and mutable

### PR DESCRIPTION
**Pull request by AndriySvyryd**
Part of #19003

### CODE REVIEW

File: src/EFCore/Infrastructure/ModelValidator.cs [hunk 1]
```diff
@@ -116,6 +116,12 @@ public virtual void Validate(IModel model, IDiagnosticsLogger<DbLoggerCategory.M
                             navigation.Name, navigation.DeclaringEntityType.DisplayName()));
                     }
 
+                    if (navigation.ForeignKey == null)
```
Comment (smitpatel) [inline src/EFCore/Infrastructure/ModelValidator.cs]: navigation -> skipNavigation?

File: src/EFCore/Metadata/Internal/SkipNavigation.cs [hunk 2]
```diff
@@ -36,21 +38,53 @@ public class SkipNavigation : PropertyBase, IMutableSkipNavigation, IConventionS
...
             [NotNull] EntityType targetEntityType,
-            [NotNull] ForeignKey foreignKey,
+            [CanBeNull] ForeignKey foreignKey,
             bool collection,
             bool onPrincipal,
             ConfigurationSource configurationSource)
             : base(name, propertyInfo, fieldInfo)
         {
-            Check.NotNull(foreignKey, nameof(foreignKey));
+            Check.NotNull(targetEntityType, nameof(targetEntityType));
```
Comment (smitpatel) [inline src/EFCore/Metadata/Internal/SkipNavigation.cs]: null check for declaringEntityType?

File: src/EFCore/Metadata/Internal/SkipNavigation.cs [hunk 3]
```diff
@@ -145,14 +187,87 @@ public virtual bool UpdateConfigurationSource(ConfigurationSource configurationS
...
 
         /// <summary>
         ///     This is an internal API that supports the Entity Framework Core infrastructure and not subject to
         ///     the same compatibility standards as public APIs. It may be changed or removed without notice in
         ///     any release. You should only use it directly in your code with extreme caution and knowing that
         ///     doing so can result in application failures when updating to a new Entity Framework Core release.
         /// </summary>
-        public virtual IConventionSkipNavigation SetInverse([CanBeNull] SkipNavigation inverse, ConfigurationSource configurationSource)
+        public virtual void UpdateForeignKeyConfigurationSource(ConfigurationSource configurationSource)
+            => _foreignKeyConfigurationSource = _foreignKeyConfigurationSource.Max(configurationSource);
```
Comment (smitpatel) [inline src/EFCore/Metadata/Internal/SkipNavigation.cs]: _foreignKeyConfigurationSource could be null.

Comment (AndriySvyryd) [reply src/EFCore/Metadata/Internal/SkipNavigation.cs]: Yes, `Max` is one of those extension methods that can be called on `null`

**Pull request merged**